### DISCUSSION TITLE
Call to a member function getDateEtHeure() on array

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46391] [Q&A] [answered]
Hello everyone 
I created a custom query in repo but when i call it in my controller it keep display  **Call to a member function getDateEtHeure() on array error**  here is my code
the query
![image](https://user-images.githubusercontent.com/101296742/169009893-bb9a4b62-78b2-427f-aaaa-b427c1f65850.png)
function de call the query
![image](https://user-images.githubusercontent.com/101296742/169009941-f9c98210-d4c5-464d-86d9-d7dea231fe59.png)
controller
![image](https://user-images.githubusercontent.com/101296742/169010006-1eca9228-6e5d-4f85-acb3-5519b473801c.png)

**Comment by DylanKas** [comment, marked as answer]
In the first case scenario when `id === 0` you do not receive the same kind of data from your repository.

`id === 0` return an array of array
`id === 1 && id === 2` return an array of `HistoriqueSondeValeur` objects (entities)

So you cannot call your method `getDateEtHeure()`.

You could modify your repository method to not get an array of array by using getResult()

So instead of 

`return $qb->getQuery()->getArrayResult();`

Use:

`return $qb->getQuery()->getResult();`

And your issue should be fixed.

**Reply by Elmatbaisaad** [reply]
thank you it works
